### PR TITLE
[various] Update flutter.dev links to more reliable destinations

### PR DESCRIPTION
**Pull request by parlough**
Contributes to https://github.com/flutter/website/issues/10363 by updating to the latest links and updating some links to use new `/to/` links that can be updated as needed.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: .ci/legacy_project/all_packages/android/.gitignore [hunk 1]
```diff
@@ -7,5 +7,5 @@ gradle-wrapper.jar
 GeneratedPluginRegistrant.java
 
 # Remember to never publicly share your keystore.
-# See https://flutter.dev/docs/deployment/android#reference-the-keystore-from-the-app
+# See /to/reference-keystore
```
Comment (stuartmorgan) [inline .ci/legacy_project/all_packages/android/.gitignore:10]: This lost the beginning of the link.

File: packages/camera/camera/README.md [hunk 2]
```diff
@@ -19,7 +19,12 @@ A Flutter plugin for iOS, Android and Web allowing access to the device cameras.
 
 ## Installation
 
-First, add `camera` as a [dependency in your pubspec.yaml file](https://flutter.dev/using-packages/).
+First, add `camera` as a
+dependency in your `pubspec.yaml` file:
+
+```bash
+flutter pub add camera
+```
```
Comment (stuartmorgan) [inline packages/camera/camera/README.md:27]: I've just been removing these blocks as I come across them; pub.dev has a prominent tab with installation instructions. So the old line can be deleted without replacement.

Same throughout the PR.

File: packages/camera/camera/README.md [hunk 3]
```diff
@@ -17,9 +17,7 @@ A Flutter plugin for iOS, Android and Web allowing access to the device cameras.
 * Record video.
 * Add access to the image stream from Dart.
 
-## Installation
-
-First, add `camera` as a [dependency in your pubspec.yaml file](https://flutter.dev/using-packages/).
```
Comment (domesticmouse) [inline packages/camera/camera/README.md:22]: I'm curious as to why this is being deleted

**Review by domesticmouse — APPROVED** [review]

File: packages/flutter_plugin_android_lifecycle/README.md [hunk 4]
```diff
@@ -13,10 +13,6 @@ major version of the Android `Lifecycle` API they expect.
 |-------------|---------|
 | **Support** | SDK 16+ |
 
-## Installation
-
-Add `flutter_plugin_android_lifecycle` as a [dependency in your pubspec.yaml file](https://flutter.dev/using-packages/).
-
```
Comment (ditman) [inline packages/flutter_plugin_android_lifecycle/README.md:19]: There's several of these where we're removing the `using-packages` link, is this intentional?

Comment (parlough) [reply packages/flutter_plugin_android_lifecycle/README.md:19]: I dropped this in response to @stuartmorgan's mention in https://github.com/flutter/packages/pull/6963#discussion_r1648273732 that this section isn't super useful with pub.dev surfacing installation information. Perhaps I misunderstood how much I should remove though.

File: packages/camera/camera/README.md (hunk 3 above)
Comment (parlough) [reply packages/camera/camera/README.md:22]: I dropped this in response to Stuart's suggestion in https://github.com/flutter/packages/pull/6963#discussion_r1648273732 that this section isn't super useful with pub.dev surfacing installation information. Perhaps I misunderstood how much I should remove though.

Comment (domesticmouse) [reply packages/camera/camera/README.md:22]: Sgtm

File: packages/flutter_plugin_android_lifecycle/README.md (hunk 4 above)
Comment (stuartmorgan) [reply packages/flutter_plugin_android_lifecycle/README.md:19]: These changes are what I had in mind, thanks!

**Review by gspencergoog — APPROVED** [review]

**Review by reidbaker — APPROVED** [review]

**Review by jmagman — APPROVED** [review]

**Review by chunhtai — APPROVED** [review]
LGTM

**Review by ditman — APPROVED** [review]
I think the changes in root `README.md` files (and some sources) are triggering the repo checks (because those should be versioned so they get published to pub).

(Not sure if we want to land this as is, to minimize incoming conflicts, and then rely on next publishes to surface the changes, or if we want to roll versions of all of modified pkgs!)

**Review by Hixie — APPROVED** [review]
LGTM

**Comment by stuartmorgan** [comment]
> (Not sure if we want to land this as is, to minimize incoming conflicts, and then rely on next publishes to surface the changes, or if we want to roll versions of all of modified pkgs!)

If there's no plan to break any of these links any time soon, we can bypass versioning.

**Review by cbracken — APPROVED** [review]

**Pull request merged**